src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -515,7 +515,7 @@ private DType(DKind kind, string namedValueKind)
             AssertValid();
         }
 
-        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null)
+        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null, HashSet<IExternalTabularDataSource> associatedDataSources = null)
```

Comment:
@LucGenetier , Where are you using this override? Just curious to know the scenario.

Reply:
This one is used in ToRecord/ToTable so that we can keep associatedDataSources when calling such methods